lib/sycamore/sycamore/docset.py

```diff
@@ -987,6 +1011,144 @@ def llm_query(self, query_agent: LLMTextQueryAgent, **kwargs) -> "DocSet":
         queries = LLMQuery(self.plan, query_agent=query_agent, **kwargs)
         return DocSet(self.context, queries)
 
+    def top_k(
+        self,
+        llm: LLM,
+        field: str,
+        k: Optional[int],
+        description: str,
+        descending: bool = True,
+        llm_cluster: bool = False,
+        unique_field: Optional[str] = None,
+        **kwargs,
+    ) -> "DocSet":
+        """
+        Determines the top k occurrences for a document field.
+
+        Args:
+            llm: LLM client.
+            field: Field to determine top k occurrences of.
+            k: Number of top occurrences.
+            description: Description of operation purpose.
```

Comment:
Description should be optional as it is only used if llm_cluster is used. I would change the name of this field to something more descriptive since it is important for the llm_cluster_entity operation, it is not just for human readability. Not sure what to call it.